From 05703932e5133ffd78b83013e83689ab1ee7e4b9 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 10:45:48 +0200
Subject: [PATCH] [SW-137121] gaudi3: erom pcie: created a device specific function

security efuse sys cfg offset value is not common among gaudi2
and gaudi3, so created to common function in device specific
common file to get whether secondary image disabled or not.

Change-Id: I725d007a966608118e50b4c058d3d1a471645b1e
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index 3b373cc..db7236a 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -17,7 +17,6 @@
 #include <common.h>
 #include <boot_if.h>
 #include <arcboot_export.h>
-#include <gaudi2/gaudi2_security.h>
 
 #define ARCBOOT_PRIMARY			0
 #define ARCBOOT_SECONDARY		1
@@ -111,10 +110,9 @@
 	uint32_t flash_image_offset;
 	uint32_t flash_ppboot_pcie_offset;
 	uint32_t ppboot_pcie_size;
-	uint32_t config_fuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
 
 	if ((image_ptr == ARCBOOT_SECONDARY) &&
-		(config_fuse & EFUSE_SYSCFG_DISABLE_SECONDARY_IMG)) {
+		(hl_is_efuse_syscfg_dis_secondary_img_set())) {
 		preboot_puts("preboot secondary image disabled!\n");
 		return ARCBOOT_BRINGUP_ERROR;
 	}
diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 49cde90..af6f87d 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -1061,3 +1061,10 @@
 	if (rst_src != HL_COLD_RST)
 		boot_info_copy_from_perst_mem();
 }
+
+bool hl_is_efuse_syscfg_dis_secondary_img_set(void)
+{
+	uint32_t config_efuse = sp_sram_readl(SECURITY_EFUSE_SYS_CFG_OFFSET);
+
+	return (config_efuse & EFUSE_SYSCFG_DISABLE_SECONDARY_IMG);
+}
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 7fbc47a..dd0d645 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1427,3 +1427,16 @@
 
 	return 0;
 }
+
+u32 sp_sram_readl(u32 offset)
+{
+	u32 reg = *((volatile u32 *) (((u32) (HL_SP_SRAM_BASE + offset))));
+	return reg;
+}
+
+bool hl_is_efuse_syscfg_dis_secondary_img_set(void)
+{
+	uint32_t config_efuse = sp_sram_readl(EFUSE_SYSCFG_OFFSET);
+
+	return (config_efuse & EFUSE_SYSCFG_DISABLE_SECONDARY_IMG);
+}
diff --git a/include/baremetal.h b/include/baremetal.h
index f8d75b9..b33158e 100644
--- a/include/baremetal.h
+++ b/include/baremetal.h
@@ -283,6 +283,8 @@
 void ppboot_boot_info_init(void);
 void preboot_recovery_flag(bool en);
 
+bool hl_is_efuse_syscfg_dis_secondary_img_set(void);
+
 #endif /* __ASSEMBLY__ */
 
 #endif /* COMMON_PREBOOT_H_ */
